### DISCUSSION TITLE
[3.2.0.dev1] Ideas for Help Instructions and User Interface Guidance

### DISCUSSION THREAD

**Discussion by slook** [nicotine-plus/nicotine-plus#1530] [Ideas]
The creation of some simple documentation for Nicotine+ is something that needs considering at some point.

**Comment by mathiascode** [comment]
Keep in mind that any user-facing documentation/help should be kept to a minimum, with intuitive UI and hints being preferred instead (we don't want another manual that remains outdated for 10+ years). Suggestions for lacking areas are welcome.

**Comment by slook** [comment]
> Suggestions for lacking areas are welcome.

Best practices... For example the use of a separate drive for the Public Shares folders.

Many Windows users seem to go with the default OS path 'C:\Documents and Settings\My Documents\Downloads\Nicotine\Complete\'

This creates a lot of problems with moving files with lonng filenames between the Incomplete and Complete folder, due to filesystem constraints. Performance and disk I/O also seems to be an issue for some users that can result in system instability.

Is there some a way this can be mitigated in the UI with perhaps some better share location defaults or in dialog context help tooltips etc?

If any long-winded explaination about the reason as to why is it best to keep the clients runtime and the users shares on separate storage devices, and ideally Downloads and Uploads on separate storage devices (to prevent disk thrashing), then I suggest a good location for this would be in the black parent page in `Preferences > Transfers`.

I believe that some specific guidance here could go a long way to mitigate some of the performance issues most commonly experienced by users on proprietary operating systems such as Windows, but the guidance would of course have to be platform independent.

**Comment by slook** [comment]
**Setup Assistant needs to do a slgihtly more comprehensive job**

Currently, the assistant isn't really very helpful...

- The user needs to be encouraged to setup Download / Upload limits (to improve system stability, so the existing setup page could be modified to select these.
- The user may want to decide if Logging is enabled or not (these files can become large).
- **_DONE_** The Setup Assistant could promote donation to 'Get Soulseek Privileges...' (see below)**

**Reply by mathiascode** [reply]
This needs to be thought out carefully. Nobody wants to sit through 50 pages of a setup process, they'll just exit the setup assistant eventually.

**Reply by slook** [reply]
Just add Download / Upload limits, on the same page as the Download/Share selection.

Alternatively, setup a defaults limits, to avoid OS hangs on single-drive systems with very fast internet connections.

**Reply by slook** [reply]
_Added in 3.2.0_

Currently @mathiascode , it is not obvious how to get Privileges... _I propose a new purpose for the last page of the Setup Assistant..._

The button would need a ToolTop hinting the URL https://www.slsknet.org/userlogin.php?username=

_Source image (https://www.freeiconspng.com/images/soulseek-png):_

The current text about download speeds would need to be moved to the previous page of the Setup Assistant.

**Comment by slook** [comment]
_Fixed in Nicotine+ 3.2.0 -_ **Give Privileges dialog box does not make it clear how to obtain Privileges if there are none (0 days left)**

_Buddy List > [user] > Give Privileges..._
![nicotine-plus Give Privileges dialog box](https://user-images.githubusercontent.com/88614182/132581354-3c9b4d7b-8aa8-40d2-929b-a19241d78c9a.png)
An alternate dialog in this case to provide a shortcut to https://www.slsknet.org/userlogin.php?username= would be more helpful in this scenario, if that is indeed how the function is worked, or offering more ways/buttons could help the user to better understand the Privileges system, ie: giving annonomously vs giving from logged in user, etc.

_Additionally, the 'Privileged' checkbox column in TreeView is ambiguous, is this not a N+ specific "Prioritize" feature?..._
![nicotine-plus Buddy Privileged](https://user-images.githubusercontent.com/88614182/132586373-94c3063c-afb5-443c-a340-242792413092.png)

The titles of these two distinct features seem to conflict with one another. Suggest review of the interface terminology is needed here.

**Reply by slook** [reply]
Fixed in Nicotine+ 3.2.0 see https://github.com/nicotine-plus/nicotine-plus/commit/72c06b1f51be27ad5263377731ce67b65f6948b6 and https://github.com/nicotine-plus/nicotine-plus/commit/ffdaf818e1d6af51a955de1cfb8f64b996b22dc9

**Comment by redactedscribe** [comment]
Because Search Result Filters and Download Filters was suggested as being potentially confusing by @slook, I noticed that the "Enable download filters" Preferences option has no tooltip. One could be added to help disambiguate, as well as possibly using different terminology  for the feature altogether, also suggested by @slook. E.g. "Download Exclusions", rather than "... Filters".

**Reply by slook** [reply]
I suggest the explanation to disambiguate be added in place of a daunting string which should be relegated to a tooltip(s):

> **Syntax:** Letters are case-insensitive. All Python regular expressions are supported if escaping is disabled. For simple filters, keeping escaping enabled is recommended.

Changing the name of the feature to Download Exclusions would also mean changing the "`Filtered`" status strings to "`Excluded`".

As to what the explaination should say depends upon the outcome of https://github.com/nicotine-plus/nicotine-plus/issues/1419
